src/snowflake/cli/api/utils/templating_functions.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Any, List, Optional
+
+from snowflake.cli.api.exceptions import InvalidTemplate
+from snowflake.cli.api.project.util import (
+    concat_identifiers,
+    identifier_to_str,
+    to_identifier,
+)
+
+
+class TemplatingFunctions:
```

Comment:
Why do these even live in a class? Feels like overhead with no benefit.

Reply:
From one of your previous comments, I got the idea that we want to avoid defining these in 2 places. 
- In this method, you only need to add the new function to this class.
- Alternatively, we can define the functions separately and manually in a map:
```
def id_to_str(arg):
  ....


- Alternatively, we can do the following:
templating_functions = {
  'id_to_str': id_to_str
}
```

But then there is a bit of boilerplate as we would have to define/reference the function in 2 places. Not a big deal, but let me know what you think.